src/utils/javaNameUtils.ts

```diff
@@ -50,7 +51,13 @@ export function getJavaClassName(name: string): string {
 }
 
 export function validateJavaFunctionName(name: string): string | undefined {
-    return validateJavaName(name);
+    if (isKeyword(name)) {
+        return localize('azFunc.JavaNameIsKeywordError', '\'{0}\' is a reserved keyword.', name);
+    }
+    if (!javaFunctionRegex.test(name)) {
```

Comment:
Don't copy this code. It's the same as the other languages and should only be in createFunction.ts

Reply:
Emm, I leave this code here is because currently the maven plugin use the function name to be the Java class name as well. Which means the java function name should both follow the function naming rule and Java class nameing rule (they are different since Java class not allow '-' in it). 

I can refactor this code when the maven plugin change the logic to name the java class.

Reply:
Ah - I thought the regex was the exact same. Didn't realize the hyphen was different. This is fine